### PR TITLE
[components] Reduce amount of duplicated code for buttons components

### PR DESCRIPTION
**Pull request by bjoerge**
There was quite a lot of duplicated code in the different variants of button components `DefaultButton`, `FakeButton`, and `AnchorButton` had no significant differences apart from the type of root element they used. I've created a component factory (or Higher Order Component if you want) that takes any given element and makes it look and feel like a button.

This also reverts c919f2598f65a45aa119e9de1c7f1f506c996b4f as the workaround for Firefox is better solved by making the `FileInput` look and behave like a button using the `createButtonLike()` factory function.

### CODE REVIEW

File: packages/@sanity/components/src/buttons/styles/DefaultButton.css [hunk 1]
```diff
@@ -32,6 +32,7 @@
   @nest &.notRecentlyHovered:focus:not(:hover) {
     overflow: hidden;
```
Comment (kristofferj) [inline packages/@sanity/components/src/buttons/styles/DefaultButton.css]: There shold be a closing bracket here. Build fails

Comment (bjoerge) [reply packages/@sanity/components/src/buttons/styles/DefaultButton.css]: Whaa, don't know what happened there. Will fix!

**Comment by bjoerge** [comment]
The merge error should be fixed now

**Comment by kristofferj** [comment]
Approving now. Will send a PR later with fixes for the design.

**Pull request merged**